src/pepr/operator/controllers/network/policies.ts

```diff
@@ -64,17 +69,24 @@ export async function networkPolicies(pkg: UDSPackage, namespace: string, istioM
     // Use the same port as the VirtualService if targetPort is not set
     const policyPort = targetPort ?? port;
 
+    // In ambient mode, we need to target the waypoint pods instead of the app pods
+    const waypointName = getWaypointName(pkg.metadata!.name!);
+
     // Create the NetworkPolicy for the VirtualService
     const policy: Allow = {
       direction: Direction.Ingress,
-      selector,
+      // In ambient mode, target the waypoint pods instead of the app pods if the package should use ambient waypoint
+      selector: getPodSelector(pkg, selector, waypointName),
       remoteNamespace: `istio-${gateway}-gateway`,
       remoteSelector: {
         app: `${gateway}-ingressgateway`,
       },
       port: policyPort,
       // Use the port, selector, and gateway to generate a description for VirtualService derived policies
-      description: `${policyPort}-${Object.values(selector)} Istio ${gateway} gateway`,
+      description:
+        istioMode === IstioState.Ambient && shouldUseAmbientWaypoint(pkg)
```

Comment:
nit: isn't checkin istioMode here redudant since `shouldUseAmbientWaypoint(pkg)` does that too?
```suggestion
        shouldUseAmbientWaypoint(pkg)
```

Reply:
OBE